usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -109,18 +109,69 @@ def get_total_results(keyword, sub_index, retries=3):
         return None
 
 
+def clean_sub_index(sub_index):
+    lookup = {"loans": "loans", "other": "other", "contract":
+              "contracts", "direct": "direct_payments", "grant": "grants", "insurance": "insurance"}
+    try:
+        return lookup[(sub_index)]
+    except KeyError:
+        logger.error('Unexpected Response')
+        return None
+
+
+def category_aggregation_query(keyword):
+    query = {
+            "query": {"query_string": {"query": keyword}},
+            "aggs": {
+                "award_category": {
+                    "terms": {
+                        "field": "award_category"
+                    }
+                },
+                "pulled_from": {
+                    "terms": {
+                        "field": "pulled_from"
+                    }
+                }
+            },
+            "size": 0
+        }
+    return query
+
+
+def get_total_results_by_category_aggregation(keyword, retries=3):
+    index_name = '{}*'.format(TRANSACTIONS_INDEX_ROOT)
+    query = category_aggregation_query(keyword)
+    response = es_client_query(index=index_name, body=query, retries=retries)
+    results = {key: 0 for key in award_categories}
+    pulled_from_idv = 0
+    insurance_count = 0
+    if response:
+        try:
+            pulled_from_buckets = response['aggregations']["pulled_from"]["buckets"]
+            for bucket in pulled_from_buckets:
+                if bucket["key"] == "idv":
+                    pulled_from_idv += bucket["doc_count"]
+                    print(pulled_from_idv)
```

Comment:
`print` funcs need to be removed before merge

Reply:
removed